src/packages/solid/utils/withScrolling.ts

```diff
@@ -49,40 +51,56 @@ export function withScrolling(adjustment: number = 0) {
           }
         }
         // no valid scroll index, complete auto scroll
         // this will be the case when we have no scroll (or an invalid one) and the lastItem is not shown (or selected is not shown)
         // if the (absolute value of current position  + the size of the visual portion of the row) is less than (the last Item position + the last Item size), then the last item is not shown
         // if scrolling backwards and the position of the selected item is less that the absolute value of the current position, the selected item is not shown
       } else if (
-        Math.abs(currentVal) + windowVal < lastItem.position + lastItem.size ||
-        newVal < Math.abs(currentVal)
+        Math.abs(rootPosition) + containerSize < lastItem.position + lastItem.size ||
+        selectedPosition < Math.abs(rootPosition)
       ) {
-        next = -newVal + adjustment;
+        next = -selectedPosition + adjustment;
       }
 
       // if we want to scroll based to the -x value of the selected
       // this will be the case for always, and edge when we are scrolling negatively and the current positioning does not have the selected item shown
       // if direction is negative and absolute value of current position is higher than the position we want to be at, the selected Item is not shown
     } else if (
       scroll === 'always' ||
-      (scroll === 'edge' && direct === 'negative' && Math.abs(currentVal) > newVal)
+      (scroll === 'edge' && direct === 'negative' && Math.abs(rootPosition) > selectedPosition)
     ) {
-      next = -newVal + adjustment;
+      next = -selectedPosition + adjustment;
 
       // if we want to scroll based on the size of the selected item
       // this will be the case for edge when we are scrolling positively and the current positioning does not have the selected item shown
       // if direction is positive and (absolute value of current position + the size of the visual portion of the row) is less than (the position we want to be at + the size of the selected Item), the selected Item is not shown
     } else if (
       scroll === 'edge' &&
       direct === 'positive' &&
-      Math.abs(currentVal) + windowVal < newVal + size
+      Math.abs(rootPosition) + containerSize < selectedPosition + selectedSize
     ) {
-      next = currentVal - size - gap;
+      next = rootPosition - selectedSize - gap;
+    }
+
+    // if inital load, edge scroll type, and we have a selected index we want to go to on start
+    else if (scroll === 'edge' && direct === 'none') {
+      let currentChildIndex = 0;
+      let currentChild, currentChildSize;
+      while (
+        currentChildIndex < componentRef.children.length &&
+        Math.abs(rootPosition) + containerSize < selectedPosition + selectedSize
+      ) {
+        currentChild = componentRef.children[currentChildIndex++];
+        currentChildSize =
+          (componentRef.flexDirection === 'row' ? currentChild.width : currentChild.height) ?? 0;
```

Comment:
instead of doing a check against flex direction, we could define which dimension we need to access as a string
```
const dimensionVariable = componentRef.flexDirection === 'row' ? 'width' : 'height';
```

and then just do
```
currentChild[dimensionVariable] ?? 0
```